kie-soup-maven-utils/kie-soup-maven-integration/src/main/java/org/appformer/maven/integration/embedder/MavenProjectLoader.java

```diff
@@ -102,9 +113,11 @@ public static MavenProject parseMavenPom(InputStream pomStream, boolean offline)
                                                             new DefaultArtifactHandler());
                     if (resolve(session, artifact)) {
                         artifacts.add(artifact);
+                        artifacts.addAll(collectTransitiveDependencies(session, dep));
```

Comment:
shouldn't we also alter the behavior of parseMavenPom(File pomFile, ...) ?

Reply:
At the beginning all the offline handling was in *parseMavenPom(InputStream pomStream, ...)*, hence why I focused changing only this method. However, you have a point. Perhaps @ederign or @porcelli could clarify if this change could be applied to both methods and I can refactor *parseMavenPom(File pomFile, ...) * to use *parseMavenPom(InputStream pomStream, ...)*.

Reply:
Hi Wiliam, I'm not an expert in this area, maybe Alex can help.